Snowflake.Data.Tests/UnitTests/Tools/UnixOperationsTest.cs

```diff
@@ -125,7 +182,9 @@ public void TestFailIfGroupOrOthersHavePermissionsToFileWithTomlConfigurationVal
             Syscall.chmod(filePath, (FilePermissions)filePermissions);
 
             // act and assert
-            Assert.Throws<SecurityException>(() => s_unixOperations.ReadAllText(filePath, TomlConnectionBuilder.ValidateFilePermissions), "Attempting to read a file with too broad permissions assigned");
+            if ((groupPermissions != FileAccessPermissions.GroupRead && groupPermissions != 0) &&
+                othersPermissions != FileAccessPermissions.OtherRead)
```

Comment:
Shouldn't it be more like this snippet?
```suggestion
            if ((groupPermissions & (FileAccessPermissions.GroupWrite | FileAccessPermissions.GroupExecute)) != 0 ||
                (othersPermissions & (FileAccessPermissions.OtherWrite | FileAccessPermissions.OtherExecute)) != 0)
```